python/in_toto_attestation/v1/resource_descriptor.py

```diff
@@ -0,0 +1,15 @@
+#  Wrapper class for in-toto attestation ResourceDescriptor protos.
+
+import in_toto_attestation.v1.resource_descriptor_pb2 as rdpb
+
+class ResourceDescriptor:
+    def __init__(self, rd=None):
+        self.pb = rdpb.ResourceDescriptor()
+
+        if rd:
+            self.pb.CopyFrom(rd)
+
+    def validate(self):
+        # at least one of name, URI or digest are required
+        if (not self.pb.name and not self.pb.uri and not self.pb.digest) or len(self.pb.digest) == 0:
```

Comment:
If self.pb.name is set but len(self.pb.digest) is 0, that's probably valid, but I think we'd through a value error?

(Or am I mis-reading the logic?)

Reply:
Ah... you're right. This check currently follows Statement.subject requirements. Thanks for catching this!